File: Intro_to_CS/Introduction-to-Computer-Science-and-Programming-using-Python/Week1/PSET1/problem2.py
"""
Problem 2
10/10 points (graded)
Assume s is a string of lower case characters.

Write a program that prints the number of times the string 'bob' occurs in s. For example, if s = 'azcbobobegghakl', then your program should print

Number of times bob occurs is: 2
"""

s = "azcbobobegghakl"

count = 0

for i in range(len(s)):
	if s[i] == 'b':
		if (len(s) - (i + 1)) >= 2:
			if s[i + 1] == 'o' and s[i + 2] == 'b':
				count += 1
													         
print(count)

File: Intro_to_CS/Introduction-to-Computer-Science-and-Programming-using-Python_JS_Edition/Week1/PSET1/problem2.js
/*
	Problem 2
	1 point possible (ungraded)
	Assume s is a string of lower case characters.

	Write a program that prints the number of times the string 'bob' occurs in s. For example, if s = 'azcbobobegghakl', then your program should print
*/

var count = 0;
s = "azcbobobegghakl";

for (var i = 0; i < s.length; i++) {
	if (s.charAt(i) == 'b' && s.length - i > 2) {
		if (s.charAt(i+1) == 'o' && s.charAt(i + 2) == 'b') {
			count++;
		}
	}
}

console.log("Number of times bob occurs is: " + count);
